### ISSUE TITLE
REST Split type uses incorrect data types for From and To fields

### ISSUE THREAD

**Issue by morningvera** [polygon-io/client-go#199]
**Describe the bug**

The rest API `Split` type contains incorrect data types for the `SplitFrom` and `SplitTo` fields. Currently, they're `int64` and they should be `float64`.

```golang
type Split struct {
	ExecutionDate string `json:"execution_date,omitempty"`
	SplitFrom     int64  `json:"split_from,omitempty"` // should be float64
	SplitTo       int64  `json:"split_to,omitempty"` // should be float64
	Ticker        string `json:"ticker,omitempty"`
}
```

**Expected behavior**

We should find a backwards compatible way to support float64 values for these fields.

**Additional context**

[Related PR](https://github.com/polygon-io/client-go/pull/198). This PR was merged for now but we should likely roll it back and implement a backwards compatible fix.

**Comment by morningvera** [comment]
Update on this: we discussed this and [one other case](https://github.com/polygon-io/client-go/issues/197) and decided to make breaking changes in both cases. We're doing this because in both cases, the original response types didn't work in the first place.

**Comment by morningvera** [comment]
Released under [v1.4.0](https://github.com/polygon-io/client-go/releases/tag/v1.4.0)
